src/util-base64.c

```diff
@@ -63,6 +63,21 @@ static inline int GetBase64Value(uint8_t c)
     return val;
 }
 
+/**
+ * \brief Checks if the given char in a byte array is Base64 alphabet
+ *
+ * \param Char that needs to be checked
+ *
+ * \return True if the char was Base64 alphabet, False otherwise
+ */
+bool IsBase64Alphabet(uint8_t encoded_byte)
+{
+    if (GetBase64Value(encoded_byte) < 0 && encoded_byte != '=') {
```

Comment:
Please check the following case:
Last blk of data that is being parsed in the remainder is:
`D=`,
with this patch, we do,
D -> Base64Alphabet? Yes -> add to the buffer
= -> Base64Alphabet? Yes -> add to the buffer

Now, the buffer becomes `D=`, we save this because it's not 4 bytes, we have next chunk of data come in, say, `BLAhblah` we add it the existing buffer, and, decode/process the remainder, which now is `D=BL`.

Is this correct?

With the earlier behavior, we'd be passing `DBLA` for decoding/processing the remainder.

Reply:
This is meant for
```
NA=
=Mg
==
```
which should decode into 42 and not into `34 03 20`
as is done for
```
NA==
Mg==
```

What is the correct result for `D=BL` ?

For me
```
D=
BLAhblah
```
And
`D=BLAhblah ` should get decoded the same way

Reply:
> For me
> D=
> BLAhblah
>
> And
> D=BLAhblah should get decoded the same way

I thought we were establishing that padding bytes `=` should only happen to the right most to be dealt with correctly? Are we not?

Reply:
I do not think we are

`DecodeBase64` can make `padding++` and then `bbidx++` for `D=BL`

But this is another bug/feature...

Reply:
Can we have test cases for each of the discussed scenarios?

Reply:
@inashivb how can we test ? Just have S-V tests with all these scenarios ?..
Looks like an overkill but unit tests running multiple times DecodeBase64 as mime does not look obvious to me as well...

Scenario being 
```
NA=
=Mg
==
```

Reply:
Looking at this again, I'm still unsure what to test against.
What I think:
Case 1: For a string like
`D=BLAHblah`, we should skip over `=` as it's not acting as a padding byte here but as an invalid char instead.
This would require fixing our Base64Decode legacy code.
Q: 
Case 2: For buf input like
```
NA=
=Mg
==
```
It is fair to expect it to be decoded like: `NA==` followed by `Mg==` as the padding bytes are where they should be.

What do you think the output should be for these cases? Will create patch and/or tests accordingly.

Reply:
Case 1 `D=BLAHblah` is beyond the scope of this PR, to see after release
Behavior was there before and will be there after

```
NA=
=Mg
==
```

should be handled as `NA==Mg==` whatever the linefeeds